config/task_test.go

```diff
@@ -901,6 +901,19 @@ func TestTaskConfig_FinalizeValidate(t *testing.T) {
 			},
 			true,
 		},
+		{
+			"valid: validate twice with services condition",
+			&TaskConfig{
+				Name:   String("task_a"),
+				Module: String("path"),
+				Condition: &ServicesConditionConfig{
+					ServicesMonitorConfig: ServicesMonitorConfig{
+						Names: []string{"serviceA"},
+					},
+				},
+			},
+			true,
+		},
```

Comment:
The tests currently fail because I'm trying to add the functionality where it's possible to get a task and use its config as the base for updating the task (pseudocode)

```go
conf, err := ctrl.Task(name)
conf.Enabled = false
_, err = ctrl.UpdateTask(conf)
```

```
error validating `condition "services"` block: regexp and names fields cannot both be unconfigured. If both are needed, consider including the list of names as part of the regex or creating separate tasks
```

However, what I found is that the config returned is considered invalid for `condition "services"` because the first validation sets `regexp` to an empty string, and the second will error that `names` and `regexp` are set.

Reply:
cc @lornasong i think it's worthwhile now to allow the rexep value to be nil :/

Reply:
@findkim thanks for letting me know. I see what you mean about the behavior being not-idempotent. In case it would be useful, I went ahead and made a draft PR to keep `regexp` set as nil: https://github.com/hashicorp/consul-terraform-sync/pull/590. If it's not what you had in mind, no worries. Let me know the best way to proceed.

Reply:
Thanks @lornasong for going ahead and making those changes! It looks great and happy to move forward with your PR so it can be idempotent but I don't see it as blocking this PR -- thoughts on this below:

Thinking through the `TaskUpdate` method signature more, I think what we currently have is the better approach with the update config instead of passing the same config object since it's likely we will not want to allow updating all options. I left a TODO note for us to determine in the future where (which package) that update config object should be added to. And for the sake of the refactor to not introduce new things, it'll use `config.TaskConfig` for now and it's now the variable is just conventionally name "updateConf"

Reply:
Sounds good. I'll go ahead with the PR to set `regexp` as nil.

Thanks for the context. Agree, I originally created the `UpdateTaskConfig` struct with the thought of preventing limiting what can be updated. Your plan to use the `config.TaskConfig` for now makes sense to me